homeassistant/components/thermostat/zwave.py

```diff
@@ -116,6 +116,8 @@ def update_properties(self):
             temps.append(int(value.data))
             if value.index == self._index:
                 self._target_temperature = value.data
+                if self._unit is None:
```

Comment:
Zwave components report in their set value, so unit_of_measurement should be collected from the device. The Frontend will convert if the system unit is the something else.